.conan/profiles/packages/settings

```diff
@@ -0,0 +1,8 @@
+{% set compiler, version = detect_api.detect_compiler() %}
+{% set os = detect_api.detect_os() %}
+{% if ((compiler == "gcc" or compiler == "clang") and os == "Linux") -%}
```

Comment:
We are duplicating the logic now in `checkMoldIsSupported` in the `conanfile.py`.  I wonder if there is any way we can inject the functionality in here to avoid duplicating the logic here?

Reply:
I tried to investigate the possibility [here](https://docs.conan.io/2/reference/config_files/profiles.html#profile-rendering) but it seems there is nothing available. There is one profile rendering function `os.getenv("ENV_VAR")` but it is not good for what we want to do because it reads system environment variables and nor in `conanfile.py` nor in `CMake` is possible to set system environment variables (if I am not wrong). I asked also in Cpplang slack #conan channel and at least for the moment it seems not possible to propagate our variable there.

Reply:
That makes sense.  Thanks for looking into it!  Let's leave as it for now and revisit in the future.  One thought is is might be ok to just restrict it to Linux and not worry about the compiler, but we'll hopefully know more once we get the CI to build the Intel compiler under Linux.

Reply:
I agree, it's cleaner if we just restrict it to Linux. Thanks.